Assets/Mirror/Editor/Weaver/CompilationFinishedHook.cs

```diff
@@ -47,6 +47,29 @@ static void HandleError(string msg)
         static void OnInitializeOnLoad()
         {
             CompilationPipeline.assemblyCompilationFinished += OnCompilationFinished;
+
+            // the first time we load the weaver we need to weave existing assemblies
+            // and reload them.
+            // Store an environment variable that will survive assembly reloads
+            // so that we know if we have run the weaver before
+            if (Environment.GetEnvironmentVariable("MIRROR_WEAVER") != "true" )
```

Comment:
how about MIRROR_WEAVED? 
MIRROR_WEAVER sounds like 'is mirror weaver present', similar to MIRROR which means 'is mirror present'.
_WEAVED would be more obvious :)